LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/GetPayRunInsightsUseCase.cs

```diff
@@ -0,0 +1,51 @@
+using Common.Extensions;
+using LBH.AdultSocialCare.Api.V1.Boundary.Payments.Response;
+using LBH.AdultSocialCare.Api.V1.Gateways.Payments.Interfaces;
+using LBH.AdultSocialCare.Api.V1.UseCase.Payments.Interfaces;
+using System;
+using System.Threading.Tasks;
+
+namespace LBH.AdultSocialCare.Api.V1.UseCase.Payments.Concrete
+{
+    public class GetPayRunInsightsUseCase : IGetPayRunInsightsUseCase
+    {
+        private readonly IPayRunGateway _payRunGateway;
+        private readonly IPayRunInvoiceGateway _payRunInvoiceGateway;
+
+        public GetPayRunInsightsUseCase(IPayRunGateway payRunGateway, IPayRunInvoiceGateway payRunInvoiceGateway)
+        {
+            _payRunGateway = payRunGateway;
+            _payRunInvoiceGateway = payRunInvoiceGateway;
+        }
+
+        public async Task<PayRunInsightsResponse> GetAsync(Guid payRunId)
+        {
```

Comment:
You have already payrun in here, no need to go db for each metrics

Reply:
The pay run here has no detail. No navigation property included

Reply:
That's right. You can get payrun with child tables and no need to go db ;) 

query for each time is more costly than getting big data volume.